blob/blob.go

```diff
@@ -598,10 +612,12 @@ func (b *Bucket) NewWriter(ctx context.Context, key string, opts *WriterOptions)
 		dopts.Metadata = md
 	}
 	ctx, cancel := context.WithCancel(ctx)
-	tctx := trace.StartSpan(ctx, "gocloud.dev/blob.NewWriter")
+	tctx := b.tracer.Start(ctx, "NewWriter")
+	end := func(err error) { b.tracer.End(tctx, err) }
+
 	defer func() {
 		if err != nil {
```

Comment:
Is `end` only called with non-nil err here because you don't want to count writes that failed right away (e.g., bad ContentType)? Is it OK to `Start` the span but then never `End` it?

Reply:
I _do_ want to count writes that failed right away.

I'm sure nothing will break horribly if I don't close a span, but why not close them all? This call to `end` plus the one in `Close` should do that.

BTW, this is old code (except for the change to `end`) that was already there. Not that we can't revisit, of course.

Reply:
Sorry, I was misreading the code, this is (still) fine.

Can you add a comment here like 

```
// If err == nil, we handed the end closure off to the
// returned *Writer; it will be called when the Writer
// is Closed.
```

to make this more obvious? Thanks!

Reply:
Done.